README


This archive contains the distribution MooseX-Constructor-AllErrors,
version 0.024:

  Capture all constructor errors

This software is copyright (c) 2009 by Hans Dieter Pearcey.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.013.



Changes
Revision history for MooseX-Constructor-AllErrors

0.024     2014-03-11 02:20:42Z
  - restore params hash to call to throw_error (RT#93709)

0.023     2014-01-22 04:01:40Z
  - use $meta->throw-error when there is an error_class (pre-Moose-2.1100)

0.022     2013-10-31 05:59:07Z
  - compatibility updates for Moose::Exception objects new in Moose 2.1200
    (ether)
  - repository migrated from shadowcat to the github moose organization

0.021     2012-06-21 22:01:04Z
  - use of auto_deref in MooseX::Constructor::AllErrors::Error::Constructor
    attributes is replaced by the Array native trait (ether)
  - new error class for 'Misc' errors has been experimentally added (ether)

0.020     2012-04-10 17:02:51Z
  - add explicit prereq for Test::CheckDeps, until the dzil plugin is fixed
    (ether)

0.019     2012-04-01 01:18:48Z
  - remove inlined versions of BUILD - it's useless to perform any checks
    *after* construction. BUILD is not the same as BUILDARGS! (ether)

0.018_001 2012-03-23
  - avoid reporting errors for undef attributes when MooseX::UndefTolerant is
    also being employed (ssoriche)

0.017   2012-01-23
  - Errors are always returned consistent with attributes' insertion_order.
    (ether)

0.016   2011-04-26
  - Remove warning from new test; added repository meta-information. (ether)

0.015   2011-04-26
  - Fix BUILD methods not being called in immutable classes (ether)

0.014   2011-04-26
  - Fix test suite for Moose error message changes (doy)

0.013   2011-02-09
  - The test suite now uses Test::Fatal instead of Test::Exception. (ether)
  - Use modern Moose APIs, to avoid warnings with Moose 1.09 (ether)
  - fix documentation mixup in
    MooseX::Constructor::AllErrors::Error::Constructor (ether)
  - moose inlining changes (doy)

0.012   2010-07-19
  - fix for moose deprecation stuff

0.011   2010-06-15
  - fix test for moose doublefail stuff

0.010   2009-01-17
  - dep on Test::Exception, since Moose only test_requires

0.009   2009-12-20
  - document the error classes

0.008   2009-12-19
  - fix constructor trait using find_type_constraint without importing it
  - don't fail if the value given for an attribute would pass validation after
    coercions

0.007   2009-04-27 15:26:35
  - avoid Moose deprecation warning

0.006   2009-04-05
  - switch from deprecated method name compute_all_applicable_attributes
  - fix Changes to be latest-first
 
0.005   2009-03-01
  - add constructor metarole for inlining

0.004   2009-03-01
  - INCOMPATIBLE CHANGE: using AllErrors no longer uses Moose implicitly
  - implementation wrapping BUILDARGS instead of hooking into several different
    parts of object construction

0.003   2009-02-04
  - avoid compile-time death

0.002   2009-02-04
  - rethrow errors that aren't from us

0.001   2009-02-03
  - the "is this crazy?" release


Makefile.PL
# This Makefile.PL for MooseX-Constructor-AllErrors was generated by
# Dist::Zilla::Plugin::MakeMaker::Fallback 0.007
# and Dist::Zilla::Plugin::MakeMaker::Awesome 0.20.
# Don't edit it but the dist.ini and plugins used to construct it.

use strict;
use warnings;

BEGIN {
my %configure_requires = (
    'ExtUtils::MakeMaker' => '6.30',
    'Module::Build::Tiny' => '0.035',
);

my @missing = grep {
    ! eval "require $_; $_->VERSION($configure_requires{$_}); 1"
} keys %configure_requires;

if (not @missing)
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";
}
else
{
    $ENV{PERL_MM_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

If you're installing manually, please retrain your fingers to run Build.PL
when present instead.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are doing something overly clever, in which case you
should consider setting the 'prefer_installer' config option in CPAN.pm, or
'prefer_makefile' in CPANPLUS, to 'mb" and '0' respectively.

You can also silence this warning for future installations by setting the
PERL_MM_FALLBACK_SILENCE_WARNING environment variable.

EOW

    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));
}
}

use 5.006;

use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Capture all constructor errors",
  "AUTHOR" => "Hans Dieter Pearcey <hdp\@cpan.org>",
  "BUILD_REQUIRES" => {},
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30",
    "Module::Build::Tiny" => "0.035"
  },
  "DISTNAME" => "MooseX-Constructor-AllErrors",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "MooseX::Constructor::AllErrors",
  "PREREQ_PM" => {
    "Moose" => 0,
    "Moose::Exporter" => 0,
    "Moose::Role" => 0,
    "Try::Tiny" => 0,
    "namespace::autoclean" => 0,
    "namespace::clean" => 0,
    "overload" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec::Functions" => 0,
    "List::Util" => 0,
    "Moose::Util::TypeConstraints" => 0,
    "Test::Fatal" => 0,
    "Test::Moose" => 0,
    "Test::More" => 0,
    "strict" => 0,
    "version" => 0,
    "warnings" => 0
  },
  "VERSION" => "0.024",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "ExtUtils::MakeMaker" => 0,
  "File::Spec::Functions" => 0,
  "List::Util" => 0,
  "Moose" => 0,
  "Moose::Exporter" => 0,
  "Moose::Role" => 0,
  "Moose::Util::TypeConstraints" => 0,
  "Test::Fatal" => 0,
  "Test::Moose" => 0,
  "Test::More" => 0,
  "Try::Tiny" => 0,
  "namespace::autoclean" => 0,
  "namespace::clean" => 0,
  "overload" => 0,
  "strict" => 0,
  "version" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



